engine/lib/mySQLRepository.js

```diff
@@ -20,6 +22,15 @@ class DatabaseRepository {
     }
   }
 
+  static createInstance(databaseConfiguration, invLoader) {
```

Comment:
i also think you can / should lazy load the repository, unless the intent was to overwrite instance.

i think you should combine createInstance & getInstance into GetOrCreateInstance().

this way only one method returns instances and the interaction is much simpler from the calling classes. there is an implicit order of operations here, since the createInstance needs to be called first. i also think the lazy load addition also fits comfortably into such a method.

Reply:
That's what I wanted to do, I just ran into the issue of always needing the app config on hand to get the instance of the mySQLRepository. Maybe it's easier to pass around than I thought, I will take another look

Reply:
feel free to ignore combining the methods if the lift is too much, i wasn't paying attention to the function arguments.

Reply:
You were on the right track! I was able to get it to work in the right way. Take a look at the new commit and let me know what you think